db-engine/src/main/java/com/splicemachine/db/iapi/types/XML.java

```diff
@@ -1148,6 +1148,11 @@ public void read(UnsafeArrayData unsafeArrayData, int ordinal) throws StandardEx
         xmlStringValue = new SQLChar(unsafeArrayData.getUTF8String(ordinal).toString());
     }
 
+    @Override
+        public int getUnsafeArrayElementSize() {
+            return 8;
+    }
+
     /**
```

Comment:
can you document how to determine the value of getUnsafeArrayElementSize? Why it is 8 for XML and UserType?

Reply:
This is the fix from @msirek related to unsaferow. I cherry-pick it from Brandon's branch about hdp 3.1 and Spark 2.4. To be honest I don't quite understand it either.

Reply:
We need to investigate how used and why the value 8 was chosen, but that should not hold up this PR.  Is this a Spark 2.4 issue?  HBase 2.0?  Something else?

I will create a Jira for this for Q4 S3.